ibis/tests/all/test_client.py

```diff
@@ -160,40 +175,103 @@ def test_nullable_input_output(con, backend, temp_table):
         ]
     )
 
-    con.create_table(temp_table, schema=sch)
+    con.create_table(table_name, schema=sch)
 
-    t = con.table(temp_table)
+    t = con.table(table_name)
 
     assert t.schema().types[0].nullable
     assert not t.schema().types[1].nullable
     assert t.schema().types[2].nullable
 
 
+@pytest.mark.parametrize('force', [False, True])
+def test_drop_table(con, table_name, schema, force):
+    _drop(
+        con,
+        name=table_name,
+        method_name='table',
+        drop_kwargs={'force': force},
+        create_kwargs={'schema': schema},
+    )
+
+
+def test_truncate_table(con, table_name):
+    con.create_table(table_name, schema=con.get_schema('functional_alltypes'))
+    con._execute(
+        "INSERT INTO {} SELECT * FROM functional_alltypes".format(table_name)
+    )
+
+    db = con.database()
+    table = db.table(table_name)
+
+    df_before, schema_before = table.execute(), table.schema()
+    con.truncate_table(table_name)
+    df_after, schema_after = table.execute(), table.schema()
+
+    assert con.exists_table(table_name)
+    assert schema_before == schema_after
+    assert df_before.shape[0] != 0 and df_after.shape[0] == 0
+
+
 # view tests
 
 
 @pytest.mark.xfail_unsupported
 @pytest.mark.xfail_backends([PySpark, Spark])
-def test_create_drop_view(con, backend, temp_view):
-    # pyspark and spark skipt because table actually is a temporary view
-    if not hasattr(con, 'create_view') or not hasattr(con, 'drop_view'):
-        pytest.xfail(
-            '{} backend doesn\'t have create_view or drop_view methods.'
-        )
+@pytest.mark.parametrize(
+    'expr',
+    [
+        [],
+        ['invalid_collumn_name'],
+        ['index', 'invalid_collumn_name'],
+        [
+            'index',
+            'id',
+            'bool_col',
+            'tinyint_col',
+            'float_col',
+            'double_col',
+            'string_col',
+            'timestamp_col',
+        ],
+    ],
+)
+def test_create_view(con, view, alltypes, expr):
+    df_alltypes = alltypes.execute()
+
+    # if list with selected cols contains invalid names
+    # 'create_view' should raise exception
+    try:
+        con.create_view(view, alltypes[expr])
+    except com.IbisTypeError:
+        assert not set(expr).issubset(df_alltypes.columns)
```

Comment:
instead of try/excepting, can you differeniate the list into those which work and those which are supposed to fail.